File: app/admin/exam_schedules.js
// app/assets/javascripts/exam_schedules.js

document.addEventListener("DOMContentLoaded", function() {
  $('#exam_schedule_program_id').on('change', function() {
    let programId = $(this).val();

    // Fetch and populate courses based on selected program
    $.ajax({
      url: '/admin/courses',
      data: { q: { program_id_eq: programId } },
      success: function(data) {
        let options = '<option value="">Select Course</option>';
        data.forEach(function(course) {
          options += `<option value="${course.id}">${course.name}</option>`;
        });
        $('#exam_schedule_course_id').html(options);
        $('#exam_schedule_section_id').html('<option value="">Select Section</option>'); // Reset sections
      }
    });
  });

  $('#exam_schedule_course_id').on('change', function() {
    let courseId = $(this).val();

    // Fetch and populate sections based on selected course
    $.ajax({
      url: '/admin/sections',
      data: { q: { course_id_eq: courseId } },
      success: function(data) {
        let options = '<option value="">Select Section</option>';
        data.forEach(function(section) {
          options += `<option value="${section.id}">${section.name}</option>`;
        });
        $('#exam_schedule_section_id').html(options);
      }
    });
  });
});

File: app/admin/exam_schedules.rb
ActiveAdmin.register ExamSchedule do
  permit_params :course_id, :program_id, :section_id, :exam_date, :start_time, :end_time, :exam_type, :classroom

  index do
    selectable_column
    id_column
    column :course
    column :program
    column :section
    column :exam_date
    column :start_time
    column :end_time
    column :exam_type
    column :classroom
    actions
  end

  form do |f|
    # Fetch options for Programs
   # programs = Program.all.map { |p| [p.program_name, p.id] }
   # 
   # # Handle nil cases
   # selected_program_id = params.dig(:exam_schedule, :program_id)
   # selected_course_id = params.dig(:exam_schedule, :course_id)
  #
   # # Fetch courses based on selected program or default to empty array
   # courses = selected_program_id.present? ? Course.by_program(selected_program_id).map { |c| [c.course_title, c.id] } : []
  #
   # # Fetch sections based on selected course or default to empty array
   # sections = selected_course_id.present? ? Section.by_course(selected_course_id).map { |s| [s.section_full_name, s.id] } : []
  
    f.inputs do
      f.input :program, as: :select, collection: Program.all.pluck(:program_name, :id)
      f.input :course, as: :select, collection: [] # Start with empty options
      f.input :section, as: :select, collection: [] # Start with empty options
      f.input :day_of_week
      f.input :start_time
      f.input :end_time
      f.input :classroom
      f.input :class_type
      f.input :instructor_name
    end
    f.actions
  end
  
  #controller do
  #  before_action :include_custom_js, only: [:new, :edit]
#
  #  private
#
  #  def include_custom_js
  #    @javascript = "exam_schedules"
  #  end
  #end
end

